### PR TITLE
Fix log10 for float32

### PR DESCRIPTION
**Pull request by Micheladar**

### CODE REVIEW

**Comment by mthrok** [comment]
@vincentqb @cpuhrsch We got a very interesting result. Using `float64` make the test result closer but still not within the range of the original test tolerance.

```
actual = tensor([[-24.5305, -27.5407, -57.3955,  ..., -57.3955,  -9.3496,  -5.2270],
        [-22.7423, -24.7428, -28.5632,  .....3955,  ..., -57.3955, -51.0177, -45.8937],
        [-57.3955, -57.3955, -57.3955,  ..., -57.3955, -52.8348, -46.1031]])
expected = tensor([[-24.5304, -27.5407, -57.3955,  ..., -57.3955,  -9.3496,  -5.2270],
        [-22.7423, -24.7429, -28.5633,  .....3955,  ..., -57.3955, -51.0175, -45.8940],
        [-57.3955, -57.3955, -57.3955,  ..., -57.3955, -52.8340, -46.1029]])
rtol = 1e-05, atol = 0.005, equal_nan = True
msg = 'Not within tolerance rtol=1e-05 atol=0.005 at input[95, 1061] (-57.3322639465332 vs. -57.342262268066406) and 1 other locations (0.00%)
```

File: test/test_librosa_compatibility.py [hunk 1]
```diff
@@ -244,7 +244,7 @@ def test_basics2(self):
         _test_compatibilities(**kwargs)
 
     # NOTE: Test passes offline, but fails on TravisCI (and CircleCI), see #372.
-    @unittest.skipIf('CI' in os.environ, 'Test is known to fail on CI')
+    # @unittest.skipIf('CI' in os.environ, 'Test is known to fail on CI')
```
Comment (vincentqb) [inline test/test_librosa_compatibility.py:247]: nit: not relevant here, but this could be replaced by `xfail` :)

File: torchaudio/functional.py [hunk 2]
```diff
@@ -392,7 +392,10 @@ def amplitude_to_DB(
     Returns:
         Tensor: Output tensor in decibel scale
     """
-    x_db = multiplier * torch.log10(torch.clamp(x, min=amin))
+    # casting to float64 is necessary because log10 behaves differently for
+    # different processors when the argument is float32. Make sure the result
+    # is a tensor of the original type.
+    x_db = multiplier * torch.log10(torch.clamp(x, min=amin).to(torch.float64)).to(x.dtype)
```
Comment (vincentqb) [inline torchaudio/functional.py:398]: what if you do not convert to `float32` back immediately but at the end?

File: torchaudio/functional.py [hunk 3]
```diff
@@ -392,7 +392,10 @@ def amplitude_to_DB(
     Returns:
         Tensor: Output tensor in decibel scale
     """
-    x_db = multiplier * torch.log10(torch.clamp(x, min=amin))
```
Comment (vincentqb) [inline torchaudio/functional.py:395]: this one isn't converted to `float64`?

**Comment by Micheladar** [comment]
I changed the function now to do the recasting at the end of the calculations.

**Pull request closed without merging**